recipe/meta.yaml

```diff
@@ -11,12 +11,11 @@ package:
 
 source:
   url: https://github.com/dmlc/xgboost/releases/download/v{{ version }}/xgboost-{{ version }}.tar.gz
-  sha256: a47ca87f3345d2a866cd7ec40933564aa2b7250592d0e9bf613951630fc1fa6f
+  sha256: a84fc7d9846c24659a2ad16788a7eefa9640b19eea9bbc65f30e0a9d53c52453
   patches:
     # xgboost patches
     - patches/0001-Enable-latest-libcxx-on-MacOS.patch  # [osx]
-    - patches/0002-Remove-nvidia-nccl-cu12-from-pyproject.toml.patch
```

Comment:
It looks like this may already be applied for us in the archive we use. Perhaps related to PR: https://github.com/dmlc/xgboost/pull/10603

Though maybe I'm missing something

cc @hcho3

Reply:
No, we still need this patch.

Reply:
Thanks for taking a look Hyunsu! 🙏

Unfortunately this doesn't make sense

Here is the URL we are using to get the source: https://github.com/dmlc/xgboost/releases/download/v2.1.2/xgboost-2.1.2.tar.gz

When looking at that source, am seeing the following for `dependencies`

```toml
dependencies = [
    "numpy",
    "scipy",
]
```

Note `nccl` is not present. As a result this patch doesn't appear to apply (nor would it be needed)

That said, please do double check. Do want to make sure we are handling this correctly

Reply:
Hmm, I just looked at the content of xgboost-2.1.2.tar.gz and it doesn't have `nccl` in the pyproject file, unlike [this one](https://github.com/dmlc/xgboost/blob/f1990391e0eb401bc1aa1309dd9f86085749c4a3/python-package/pyproject.toml#L34). I suppose it was manually patched by @trivialfis

Reply:
Is this [the code]( https://github.com/dmlc/xgboost/blob/f1990391e0eb401bc1aa1309dd9f86085749c4a3/dev/release-artifacts.py#L121-L142 ) that is used to create that artifact?

If so, it appears to [patch out `nccl` from the `dependencies`]( https://github.com/dmlc/xgboost/blob/8fe518600aad007dd5e50fecadd19ec44b15d235/tests/buildkite/remove_nccl_dep.patch#L9 )

Not very familiar with the upstream build process. So please correct me if I'm misunderstanding things

Reply:
Ah never mind, forgot that I wrote the script ;)